csharp/src/Ice/LoggerAdminLogger.cs

```diff
@@ -38,176 +52,35 @@ public void Warning(string message)
-                    {
-                        Monitor.Wait(this);
-                    }
-
-                    if (_destroyed)
-                    {
-                        break; // for(;;)
-                    }
-
-                    Debug.Assert(_jobQueue.Count > 0);
-                    job = _jobQueue.Dequeue();
-                }
-
-                foreach (IRemoteLoggerPrx p in job.RemoteLoggers)
+                foreach (RemoteLoggerQueue p in remoteLoggers)
                 {
-                    if (_loggerAdmin.GetTraceLevel() > 1)
-                    {
-                        _localLogger.Trace(TraceCategory, "sending log message to `" + p.ToString() + "'");
-                    }
-
-                    try
-                    {
-                        //
-                        // p is a proxy associated with the _sendLogCommunicator
-                        //
-                        p.LogAsync(job.LogMessage).ContinueWith(
-                            (t) =>
-                            {
-                                try
-                                {
-                                    t.Wait();
-                                    if (_loggerAdmin.GetTraceLevel() > 1)
-                                    {
-                                        _localLogger.Trace(TraceCategory, "log on `" + p.ToString()
-                                                           + "' completed successfully");
-                                    }
-                                }
-                                catch (AggregateException ae)
-                                {
-                                    if (ae.InnerException is CommunicatorDestroyedException)
-                                    {
-                                        // expected if there are outstanding calls during communicator destruction
-                                    }
-
-                                    Debug.Assert(ae.InnerException != null);
-                                    _loggerAdmin.DeadRemoteLogger(p, _localLogger, ae.InnerException, "log");
-                                }
-                            },
-                            System.Threading.Tasks.TaskScheduler.Current);
-                    }
-                    catch (System.Exception ex)
+                    p.Queue(async prx =>
                     {
-                        _loggerAdmin.DeadRemoteLogger(p, _localLogger, ex, "log");
-                    }
+                        if (_loggerAdmin.GetTraceLevel() > 1)
+                        {
+                            _localLogger.Trace(TraceCategory, $"sending log message to `{prx}'");
+                        }
+                        await prx.LogAsync(logMessage);
+                        if (_loggerAdmin.GetTraceLevel() > 1)
+                        {
+                            _localLogger.Trace(TraceCategory, $"log on `{prx}' completed successfully");
```

Comment:
sent successfully? It's likely all this can be move to Queue.

Reply:
Fixed